pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -304,26 +304,6 @@ func (s *Reconciler) Exists(ctx context.Context) (bool, error) {
 		return false, nil
 	}
 
-	if s.scope.Machine.Spec.ProviderID == nil || *s.scope.Machine.Spec.ProviderID == "" {
-		// TODO: This should be unified with the logic for getting the nodeRef, and
-		// should potentially leverage the code that already exists in
-		// kubernetes/cloud-provider-azure
-		providerID := fmt.Sprintf("azure:////%s", *s.scope.MachineStatus.VMID)
-		s.scope.Machine.Spec.ProviderID = &providerID
-	}
-
-	// Set the Machine NodeRef.
-	if s.scope.Machine.Status.NodeRef == nil {
-		nodeRef, err := getNodeReference(s.scope)
```

Comment:
Given `getNodeReference` is called only here, you might remove the function as well